packages/mobile/src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,362 @@
+  const onNavigationButtonPress = () => {
+    if (!dappSelected) {
+      // Should never happen
+      Logger.warn(TAG, 'Internal error. There was no dapp selected')
+      return
+    }
+    navigateToURI(dappSelected.dappUrl)
+    setBottomSheetVisible(false)
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader
+        left={<BackButton />}
+        title={t('dappsScreen.title')}
+        right={
+          <TopBarIconButton
+            icon={<Help />}
+            onPress={onHelpPress}
+            style={styles.helpIconContainer}
+          />
+        }
+      />
+      <BottomSheet isVisible={isBottomSheetVisible} onBackgroundPress={onBottomSheetClose}>
+        <View>
+          <View style={styles.bottomSheetHeader}>
+            <TopBarIconButton icon={<QuitIcon />} onPress={onBottomSheetClose} />
+          </View>
+          <View style={styles.centerContainer}>
+            <Text style={{ ...fontStyles.h2, textAlign: 'center', paddingVertical: 16 }}>
+              {t('dappsScreenBottomSheet.title', { dappName: dappSelected?.name })}
+            </Text>
+            <Text style={{ ...fontStyles.regular, textAlign: 'center', flex: 1 }}>
+              {t('dappsScreenBottomSheet.message')}
+            </Text>
+            <Button
+              style={styles.bottomSheetButton}
+              onPress={onNavigationButtonPress}
+              text={t('dappsScreenBottomSheet.button', { dappName: dappSelected?.name })}
+            />
+          </View>
+        </View>
+      </BottomSheet>
+
+      <Dialog
+        title={t('dappsScreenHelpDialog.title')}
+        isVisible={isHelpDialogVisible}
+        secondaryActionText={t('dappsScreenHelpDialog.dismiss')}
+        secondaryActionPress={() => setHelpDialogVisible(false)}
+      >
+        {t('dappsScreenHelpDialog.message')}
+      </Dialog>
+
+      <ScrollView style={styles.scrollContainer}>
+        <View style={styles.descriptionContainer}>
+          <Text style={{ ...fontStyles.h1, flex: 1 }}>{t('dappsScreen.message')}</Text>
+          <View style={styles.descriptionImage}>
+            <DappsExplorerLogo />
+          </View>
+        </View>
+        <>
+          {loading && (
+            <View style={styles.centerContainer}>
+              <ActivityIndicator
+                style={styles.loadingIcon}
+                size="large"
+                color={colors.greenBrand}
+                testID="DAppExplorerScreen/loading"
+              />
+            </View>
+          )}
+          {!loading && error && (
+            <View style={{ ...styles.centerContainer, marginTop: 96 }}>
+              <Text style={fontStyles.regular}>{t('dappsScreen.errorMessage')}</Text>
+            </View>
+          )}
+          {!loading &&
+            !error &&
+            result &&
+            result.map((category: CategoryWithDapps) => (
+              <CategoryWithDapps category={category} onDappPress={onDappPress} />
+            ))}
```

Comment:
This is ok when we have few Dapps / Categories, but will become a performance issue if we have more than 10 or so of them. At that point we should favor a `FlatList`.

Reply:
It makes sense, I will change it

Reply:
Or ˋSectionList`